tests/Doctrine/Tests/ORM/Functional/OneToOneSelfReferentialAssociationTest.php

```diff
@@ -74,9 +74,11 @@ public function testFind(): void
 
     public function testEagerLoadsAssociation(): void
     {
-        $this->createFixture();
+        $customerId = $this->createFixture();
+
+        $query = $this->_em->createQuery('select c, m from Doctrine\Tests\Models\ECommerce\ECommerceCustomer c left join c.mentor m where c.id = :id');
+        $query->setParameter('id', $customerId);
```

Comment:
What's the story about this changed test?

Reply:
The test assumed that the entities were inserted (and have their IDs assigned) in the order they were added to or discovered by the entity manager.

But, since Luke has Obi-Wan as his mentor, Obi-Wan will now be inserted first. This allows us to make the `mentor` connection without an extra `UPDATE` query.

I wrote the `TopologicalSort` class so that it keeps objects in order as much as possible (otherwise, it would have broken lots of tests), but associations and avoiding extra queries are the main reason for re-ordering inserts.